### PR TITLE
feat(sections,projects): model description on read and write types

### PR DESCRIPTION
**Pull request by lmjabreu**
## What

Models project and section `description` (26Q2 Project Essentials) on the SDK's read and write types. The backend shipped both (sections landed 2026-06-10 via [Todoist#28403](https://github.com/Doist/Todoist/pull/28403)), but the SDK didn't model the section side and omitted `description` from the write args — so the CLI and MCP currently need casts and can't read section descriptions at all. This unblocks them.

## Changes

**Sections**
- `SectionBaseSchema` gains `description: z.string().nullable()` — the backend column is `str | None` (null when unset), so REST and sync section reads now carry it.
- `WebhookSectionSchema` overrides `description` to tolerate omission (`.nullish()` → `null`), mirroring the existing `updatedAt` handling, so leaner webhook payloads still parse.
- `AddSectionArgs` gains optional `description`; `UpdateSectionArgs` makes `name` optional and adds `description?: string | null` (`null` clears, matching the backend's `NULL_CLEARS`), enabling description-only updates.

**Projects**
- `AddProjectArgs` / `UpdateProjectArgs` gain optional `description` (empty string clears on update, matching `NULL_KEEPS_UNCHANGED`). The project read schema already carried `description`, so no read change there.

## Clear-semantics note (intentional asymmetry, from the backend)

- Project update: omit `description` to keep, `""` to clear.
- Section update: omit to keep, `null` to clear.

These mirror the backend's `NULL_KEEPS_UNCHANGED` (projects) vs `NULL_CLEARS` (sections) contracts and are documented in the arg JSDoc.

## Testing

- Read parsing for `description` (string and `null`).
- Webhook tolerance: present description surfaces; missing key normalises to `null`.
- `addSection` / `updateSection` and `addProject` / `updateProject` forward `description` in the request body, including the section `null`-clear and project `""`-clear paths, plus a description-only section update (no `name`).
- Full suite green (633 tests), lint/format, build all pass. (The 4 pre-existing `ts-compile-check` errors in apps/ui-extensions tests are unrelated to this change and present on `main`.)

## Downstream

Unblocks the draft consumer PRs: CLI [todoist-cli#395](https://github.com/Doist/todoist-cli/pull/395)/[#396](https://github.com/Doist/todoist-cli/pull/396) and MCP [todoist-mcp#500](https://github.com/Doist/todoist-mcp/pull/500)/[#501](https://github.com/Doist/todoist-mcp/pull/501). Draft until reviewed; once released, bump the pins there to drop the temporary casts.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by lmjabreu** [comment]
@Doistbot /review

File: src/types/sections/requests.ts [hunk 1]
```diff
@@ -33,12 +33,19 @@ export type AddSectionArgs = {
...
+    description?: string
 }
 
 /**
  * Arguments for updating a section.
  * @see https://developer.todoist.com/api/v1/#tag/Sections/operation/update_section_api_v1_sections__section_id__post
  */
 export type UpdateSectionArgs = {
-    name: string
+    name?: string
```
Comment (doistbot) [inline src/types/sections/requests.ts]: <big><kbd>🟡 P2</kbd></big>  Making `name` optional here means `UpdateSectionArgs` now accepts `{}`. `SectionClient.updateSection()` forwards `args` directly as the POST body, so `api.updateSection(id, {})` will now type-check and send an empty update request. To preserve the old non-empty payload guarantee while still allowing description-only updates, wrap this shape in a `RequireAtLeastOne<...>`-style utility.

File: src/types/webhooks/sections.test.ts [hunk 2]
```diff
@@ -41,6 +41,23 @@ describe('section:* payloads', () => {
...
+    test('surfaces a section description when present', () => {
+        const payload = parseWebhookPayload(
+            envelope('section:updated', rawSection({ description: 'Sprint notes' })),
+        )
+        if (payload.eventName !== 'section:updated') throw new Error('expected section:updated')
+
+        expect(payload.eventData.description).toBe('Sprint notes')
+    })
+
+    test('tolerates a missing description key (normalises to null)', () => {
```
Comment (doistbot) [inline src/types/webhooks/sections.test.ts:53]: <big><kbd>🟡 P2</kbd></big>  This covers the webhook-only exception, but the suite still doesn't pin the opposite contract: non-webhook section reads must reject a missing `description`. A small sibling test via `validateSection` or a GET section response would catch regressions where `SectionBaseSchema` gets loosened to `.nullish()` and starts leaking `undefined` into the public `Section` type.

**Review by doistbot — COMMENTED** [review]
This introduces the `description` field for both projects and sections across read and write types, properly accommodating the API's update asymmetry and webhook structures.

Few things worth tightening:
- Wrap `UpdateSectionArgs` in a `RequireAtLeastOne` utility to prevent empty update requests now that `name` is optional.
- Add a strictness test to ensure non-webhook section reads actively reject payloads missing a `description`, preventing accidental schema loosening in the future.

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-sdk-typescript&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-sdk-typescript%2Fpull%2F620&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253Ae27f8590-6516-11f1-92d8-246e9f5609c3%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3Ae27f8590-6516-11f1-92d8-246e9f5609c3&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/types/sections/requests.ts (hunk 1 above)
Comment (lmjabreu) [reply src/types/sections/requests.ts]: Done. Wrapped `UpdateSectionArgs` in a new `RequireAtLeastOne` utility, so `api.updateSection(id, {})` no longer type-checks.

File: src/types/webhooks/sections.test.ts (hunk 2 above)
Comment (lmjabreu) [reply src/types/webhooks/sections.test.ts:53]: Done. Added a `validateSection` test asserting a section missing `description` is rejected (the webhook variant keeps its tolerant override).

**Comment by lmjabreu** [comment]
@Doistbot /review

File: src/types/sections/requests.ts [hunk 3]
```diff
@@ -33,12 +33,19 @@ export type AddSectionArgs = {
...
 
 /**
- * Arguments for updating a section.
+ * Arguments for updating a section. At least one field must be provided.
  * @see https://developer.todoist.com/api/v1/#tag/Sections/operation/update_section_api_v1_sections__section_id__post
  */
-export type UpdateSectionArgs = {
-    name: string
-}
+export type UpdateSectionArgs = RequireAtLeastOne<{
```
Comment (doistbot) [inline src/types/sections/requests.ts:45]: <big><kbd>🟡 P2</kbd></big>  This still allows an effectively empty update: with the current `RequireAtLeastOne` shape, `{ name: undefined }` and `{ description: undefined }` satisfy `UpdateSectionArgs`, and `request()` later `JSON.stringify`s them down to `{}`. `Required<Pick<...>>` keeps `undefined` from optional members. To preserve the non-empty payload guarantee, apply `RequireAtLeastOne` to a required base shape (`{ name: string; description: string | null }`) or strip `undefined` in the helper.

File: src/types/webhooks/sections.test.ts (hunk 2 above)
Comment (doistbot) [inline src/types/webhooks/sections.test.ts:53]: <big><kbd>🟡 P2</kbd></big>  This only locks down the omitted-key branch. `WebhookSectionSchema` also accepts `description: null`; add that case here so a future change from `.nullish()` to `.optional()` wouldn't slip through while dropping one supported payload shape.

**Review by doistbot — COMMENTED** [review]
This adds solid support for reading and writing project and section descriptions while correctly navigating the backend's asymmetric clearing contracts.

Few things worth tightening:
- The `RequireAtLeastOne` guard still allows effectively empty updates (like `{ name: undefined }`), which can be sealed by applying it to a required base shape or stripping `undefined`.
- The webhook test covers the omitted-key branch but misses explicit `null`s; adding a test case for this will prevent accidental regressions if the schema ever changes from `.nullish()` to `.optional()`.

I also included a few optional follow-up notes in the details below.

<details>
<summary>Optional follow-up note (1)</summary>

- **[P3] src/types/sections/requests.ts:44**: `RequireAtLeastOne` is the key safety net here, but the suite doesn't pin it. The repo already uses `@ts-expect-error` assertions for type contracts; add one for `api.updateSection('123', {})` (or `{}` as `UpdateSectionArgs`) so the non-empty update guarantee can't silently regress.

</details>

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-sdk-typescript&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-sdk-typescript%2Fpull%2F620&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253A520202c0-6518-11f1-9d04-f60bbbc21fd4%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3A520202c0-6518-11f1-9d04-f60bbbc21fd4&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/types/sections/requests.ts (hunk 3 above)
Comment (lmjabreu) [reply src/types/sections/requests.ts:45]: Fixed. `RequireAtLeastOne` now requires the chosen key to carry a *defined* value, so `{ name: undefined }` no longer satisfies `UpdateSectionArgs` (`null` is still allowed for the description clear). Added a `@ts-expect-error` contract test pinning that `{}` and `{ name: undefined }` are rejected.

File: src/types/webhooks/sections.test.ts (hunk 2 above)
Comment (lmjabreu) [reply src/types/webhooks/sections.test.ts:53]: Done. Added an explicit `description: null` webhook test alongside the omitted-key case, so a future `.nullish()` -> `.optional()` change would be caught.

**Comment by lmjabreu** [comment]
@Doistbot /review

**Review by doistbot — COMMENTED** [review]
This cleanly models the new `description` field for projects and sections across the SDK's read and write types, effectively handling the backend's asymmetric update contracts and webhook edge cases. No issues were flagged.

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-sdk-typescript&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-sdk-typescript%2Fpull%2F620&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253Af053b9d0-651a-11f1-9b64-5097673438a9%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3Af053b9d0-651a-11f1-9b64-5097673438a9&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/todoist-api.projects.test.ts [hunk 4]
```diff
@@ -171,6 +186,26 @@ describe('TodoistApi project endpoints', () => {
             expect(capturedBodies[0]).toEqual({ folder_id: '42' })
             expect(capturedBodies[1]).toEqual({ folder_id: null })
         })
+
+        test('forwards description, sending an empty string to clear it', async () => {
```
Comment (doistbot) [inline src/todoist-api.projects.test.ts:203]: <big><kbd>🟡 P2</kbd></big>  This covers the runtime clear path (`''`), but it doesn’t pin the new project/section asymmetry at the type level. If `UpdateProjectArgs.description` is later widened to `string | null`, `api.updateProject(id, { description: null })` would start compiling even though the backend treats `null` as "leave unchanged," not "clear." Add an adjacent `@ts-expect-error` contract test for `{ description: null }` to lock that down.

**Review by doistbot — COMMENTED** [review]
This PR cleanly models the new `description` field for projects and sections across the SDK's read and write interfaces.

Few things worth tightening:
- Pin the project clear semantics at the type level by adding a `@ts-expect-error` test for `{ description: null }`.

I also included a few optional follow-up notes in the details below.

<details>
<summary>Optional follow-up note (1)</summary>

- **[P3] src/types/common.ts:17**: This introduces a local `RequireAtLeastOne`, but the repo already depends on `type-fest` and uses its request-shape helpers in sibling files like `types/tasks/requests.ts` and `types/comments/requests.ts`. Keeping a second in-house version of the same abstraction adds maintenance cost and makes the rules harder to remember. You can preserve the current “selected key must be defined” behavior by using `type-fest`’s `RequireAtLeastOne` on a required base shape instead (for example, `RequireAtLeastOne<{ name: string; description: string | null }>` in `UpdateSectionArgs`).

</details>

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-sdk-typescript&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-sdk-typescript%2Fpull%2F620&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253Afcb615f0-651b-11f1-922f-f23d5e88b2ae%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3Afcb615f0-651b-11f1-922f-f23d5e88b2ae&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/todoist-api.projects.test.ts (hunk 4 above)
Comment (lmjabreu) [reply src/todoist-api.projects.test.ts:203]: Done. Added a `@ts-expect-error` contract test asserting `UpdateProjectArgs` rejects `{ description: null }` (projects clear with `""`; `null` would mean leave-unchanged). This pins the project/section asymmetry and would break if the type were ever widened to `string | null`.

**Comment by lmjabreu** [comment]
@Doistbot /review

**Review by scottlovegrove — APPROVED** [review]
